Kuroba/app/src/main/java/com/github/adamantcheese/chan/report/ReportProblemController.kt

```diff
@@ -0,0 +1,154 @@
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+
+class ReportProblemController(context: Context) : Controller(context) {
+    private lateinit var compositeDisposable: CompositeDisposable
+    private lateinit var reportActivityProblemTitle: TextInputEditText
+    private lateinit var reportActivityProblemDescription: TextInputEditText
+    private lateinit var reportActivityAttachLogsButton: AppCompatCheckBox
+    private lateinit var reportActivityLogsText: TextInputEditText
+    private lateinit var reportActivityCancel: AppCompatButton
+    private lateinit var reportActivitySendReport: AppCompatButton
+
+    private val presenter: ReportProblemPresenter = ReportProblemPresenter()
+    private var loadingViewController: LoadingViewController? = null
+    private var onFinishedListener: (() -> Unit)? = null
+
+    override fun onCreate() {
+        Chan.inject(this)
+
+        view = inflate(context, R.layout.activity_report).apply {
+            reportActivityProblemTitle = findViewById(R.id.report_activity_problem_title)
+            reportActivityProblemDescription = findViewById(R.id.report_activity_problem_description)
+            reportActivityAttachLogsButton = findViewById(R.id.report_activity_attach_logs_button)
+            reportActivityLogsText = findViewById(R.id.report_activity_logs_text)
+            reportActivityCancel = findViewById(R.id.report_activity_cancel)
+            reportActivitySendReport = findViewById(R.id.report_activity_send_report)
+        }
+
+        presenter.onCreate(this)
+        compositeDisposable = CompositeDisposable()
+
+        val logs = LogsController.loadLogs()
+        if (logs != null) {
+            reportActivityLogsText.setText(logs)
+        }
+
+        reportActivityAttachLogsButton.setOnCheckedChangeListener { _, isChecked ->
+            reportActivityLogsText.isEnabled = isChecked
+        }
+        reportActivityCancel.setOnClickListener { onFinishedListener?.invoke() }
+        reportActivitySendReport.setOnClickListener { onSendReportClick() }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+
+        onFinishedListener = null
+        presenter.onDestroy()
+        compositeDisposable.dispose()
+    }
+
+    fun setOnFinishedCallback(callback: () -> Unit) {
+        this.onFinishedListener = callback
+    }
+
+    private fun onSendReportClick() {
+        val title = reportActivityProblemTitle.text?.toString() ?: ""
+        val description = reportActivityProblemDescription.text?.toString() ?: ""
+        val logs = reportActivityLogsText.text?.toString() ?: ""
+
+        if (title.isEmpty()) {
+            reportActivityProblemTitle.error =
+                    context.getString(R.string.report_activity_title_cannot_be_empty_error)
+            return
+        }
+
+        if (description.isEmpty()) {
```

Comment:
It's probably fine to allow empty descriptions, but only if logs are attached.

Reply:
Yeah, that's probably a good idea.

Reply:
Done.